x-pack/plugins/observability_solution/apm/server/routes/storage_explorer/has_storage_explorer_privileges.ts

```diff
@@ -18,7 +18,13 @@ export async function hasStorageExplorerPrivileges({
   apmEventClient: APMEventClient;
 }) {
   const {
-    indices: { transaction, span, metric, error },
+    // Only use apm index patterns and ignore OTel, as the storage explorer only supports APM data
+    indices: {
+      transaction = 'traces-apm*,apm-*',
+      span = 'traces-apm*,apm-*',
+      metric = 'metrics-apm*,apm-*',
+      error = 'logs-apm*,apm-*',
```

Comment:
The root cause of the failing tests can be found by starting from here (at least that's how I found this). 

Ultimately this tests storage explorer privileges. The problem is that we want to change `apm_indices.spec.ts` by appending the OTel indices to the already existing APM indices so the UI works with OTel data as well. At this point we did not do any work for storage explorer + OTel data and I'd suggest marking Storage Explorer + OTel not supported at this point. (only reason this comes up is the failing test).

So, ultimately this `hasStorageExplorerPrivileges` should have a limited index pattern that only contains the old apm indices, but not the OTel one, because for OTel we don't set storage explorer related things.

The code above still does not work, because it only sets default values, if they are not present (at least that's my understanding with 0 TypeScript knowledge), but the caller actually passes those with the values set from `apm_indices.spec.ts` containing the OTel indices as well - and that's why the test fails, because those indices will not have storage explorer privileges.

This is called [from here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/storage_explorer/route.ts#L192-L196) and the `apmEventClient` passed into this pulls index patterns from `apm_indices.spec.ts` via `MinimalAPMRouteHandlerResources`.

So ideally, there should be 2 versions of `MinimalAPMRouteHandlerResources`: one with the old index patterns (apm only), and one with the new one (apm + OTel). For storage explorer tests we could use apm only, for the test use the new one (apm + OTel).

Full disclosure: I have absolutely no idea about TypeScript - seems like a fancy stuff in any case...

Reply:
See https://github.com/elastic/kibana/pull/190943/commits/f743aeb208fd74a65752fe32fc4044bb4ee24661 with an early stage idea on how to address this: https://github.com/elastic/kibana/pull/190943/commits/f743aeb208fd74a65752fe32fc4044bb4ee24661